### ISSUE TITLE
Option to switch off logger.error for unsuccessful notifications

### ISSUE THREAD

**Issue by cat-bro** [Fatal1ty/aioapns#15]
Each unsuccessful notification is leading to a `logger.error` (here https://github.com/Fatal1ty/aioapns/blame/master/aioapns/client.py#L51).  This could become noisy with many notifications being sent. We'd prefer to be able to handle these errors based on the response.status, on a case-by-case basis.  An example is that if the response status code is 410, we'd like to quietly deregister the token without an error notification in the system.

When adding this logging error, did you have a different token deregistering pattern in mind?
It seems like a common approach to deregistering tokens when they return a particular error code after trying to push a notification to them..

Ideally, you raise an exception here, for the user to handle.
Or maybe at least reduce the loglevel to a warning, or introduce an option to skip this logging step (so that the user can check the response status themselves)?

Disabling logging from our end would mean losing other potentially useful stuff, so we'd rather not do that.

**Comment by skabber** [comment]
@cat-bro I have a fork that does this. See #24

**Comment by cat-bro** [comment]
thank you @skabber

**Comment by alexmv** [comment]
This is also something that we just ran into -- if you want to make sure that callers handle the exceptional case, it seems like raising an exception is the more flexible option.  Unfortunately. it looks like #24 doesn't actually _prevent_ the `logging.error` call if the `err_func` is passed, so it doesn't address the issue.

@Fatal1ty, would you be amenable to a PR that raised an exception in the `not response.is_successful` case?  It'd be a breaking change, of course, but I think that it's be a more useful interface.

**Comment by Fatal1ty** [comment]
@alexmv Yes, you’re right it has to be an exception here. This project needs to be refreshed but unfortunately I don’t use it in my current projects and don’t have time to keep it up to date. Of course, I will accept a PR and release a new version if you help to make the necessary changes.

**Comment by andersk** [comment]
> Unfortunately. it looks like #24 doesn't actually _prevent_ the `logging.error` call if the `err_func` is passed, so it doesn't address the issue.

It could, actually: since `error_func` is called before the `logging.error`, you can use it to skip the `logging.error` by raising an exception yourself.
